packages/components/src/components/hds/popover-primitive/index.ts

```diff
@@ -167,27 +163,58 @@ export default class HdsPopoverPrimitive extends Component<HdsPopoverPrimitiveSi
       registerEvent(this._popoverElement, ['toggle', this.onTogglePopover]);
 
       // we need to spread the argument because if it's set via `{{ hash … }}` Ember complains when we overwrite one of its values
-      const anchoredPositionOptions: HdsAnchoredPositionOptions = {
+      this._anchoredPositionOptions = {
         ...named.anchoredPositionOptions,
       };
 
-      // Apply the `hds-anchored-position` modifier to the "popover" element
-      // (notice: this function runs the first time when the element the modifier was applied to is inserted into the DOM, and it autotracks while running.
-      // Any tracked values that it accesses will be tracked, including the arguments it receives, and if any of them changes, the function will run again)
-      // This modifiers uses the Floating UI library to provide:
-      // - positioning of the "popover" in relation to the "toggle"
-      // - collision detection (optional)
+      this._linkToggleAndPopover();
+    }
+  );
+
+  // Apply the `hds-anchored-position` modifier to the "popover" element
+  // (notice: this function runs the first time when the element the modifier was applied to is inserted into the DOM, and it autotracks while running.
+  // Any tracked values that it accesses will be tracked, including the arguments it receives, and if any of them changes, the function will run again)
+  // This modifiers uses the Floating UI library to provide:
+  // - positioning of the "popover" in relation to the "toggle"
+  // - collision detection (optional)
+  private _updatePositioning(): void {
```

Comment:
[suggestion] `_updatePositioning` reminds me of the action that the Floating UI library is performing, while here we're actually applying the modifier to the popover element
```suggestion
  private _applyAnchoredPositionModifier(): void {
```